assets/js/components/post-searcher.js

```diff
@@ -112,28 +115,37 @@ class PostSearcher extends Component {
 	}
 
 	onConfirm( selection ) {
-		this.setState( {
-			selection,
-		} );
+		const { results } = this.state;
+		let match;
+		// Check to that the selection is "valid".
+		if ( ! Array.isArray( results ) || ! this.disallowedSelections.includes( selection ) ) {
+			match = results.find( ( result ) => result.post_title === selection );
+			if ( selection && ! isUndefined( match ) ) {
+				this.setState( {
+					selection,
+					canSubmit: true,
+					match,
+				} );
+			}
+		} else {
+			this.setState( {
+				canSubmit: false,
+			} );
+		}
 	}
 
 	onClick() {
-		const { results, selection } = this.state;
-		const match = find(
-			results,
-			( result ) => {
-				return result.post_title === selection;
-			}
-		);
-
-		document.location = getSiteKitAdminURL(
-			'googlesitekit-dashboard',
-			{
-				id: match.id,
-				permaLink: match.permalink,
-				pageTitle: selection,
-			}
-		);
+		const { canSubmit, selection, match } = this.state;
+		if ( canSubmit ) {
```

Comment:
This conditional is unnecessary as this event handler won't be invoked if the button clicked when disabled.
We don't usually double-guard handlers like this when it's already covered where it's invoked.

Instead I think it would be a good idea to check that there is a valid `match` instead 👍 
```suggestion
		if ( match && match.id ) {
```

Reply:
Updated.